dotCMS/src/main/java/com/dotmarketing/db/TomcatDatasourceStrategy.java

```diff
@@ -0,0 +1,48 @@
+package com.dotmarketing.db;
+
+import com.dotcms.repackage.com.zaxxer.hikari.HikariConfig;
+import com.dotcms.repackage.com.zaxxer.hikari.HikariDataSource;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Constants;
+import com.dotmarketing.util.Logger;
+import com.liferay.util.JNDIUtil;
+import javax.naming.InitialContext;
+import javax.sql.DataSource;
+
+/**
+ * Singleton class that obtains a datasource from a <b>context.xml</b> file
+ * @author nollymar
+ */
+public class TomcatDatasourceStrategy implements DotDatasourceStrategy {
+
+    private TomcatDatasourceStrategy(){}
+
+    private static class SingletonHelper{
+        private static TomcatDatasourceStrategy INSTANCE = new TomcatDatasourceStrategy();
+    }
+
+    public static TomcatDatasourceStrategy getInstance(){
+        return SingletonHelper.INSTANCE;
+    }
+
+    @Override
+    public DataSource getDatasource() {
+        try {
+            final InitialContext ctx = new InitialContext();
+            final HikariConfig config = new HikariConfig();
+            config.setDataSource((DataSource) JNDIUtil.lookup(ctx, Constants.DATABASE_DEFAULT_DATASOURCE));
+            return new HikariDataSource(config);
+        } catch (Throwable e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)